numba/np/random/generator_methods.py

```diff
@@ -58,16 +73,166 @@ def impl(inst, size=None, dtype=np.float64):
             return nb_dt(dist_func(inst.bit_generator))
         return impl
     else:
-        assert any([isinstance(size, UniTuple) and
-                   isinstance(size.dtype, types.Integer),
-                   isinstance(size, Tuple) and size.count == 0,
-                   isinstance(size, types.Integer)]
-                   ), "Size argument either of None,"\
-            + " an integer, an empty tuple or a tuple of integers"
+        check_size(size)
 
         def impl(inst, size=None, dtype=np.float64):
             out = np.empty(size, dtype=dtype)
             for i in np.ndindex(size):
                 out[i] = dist_func(inst.bit_generator)
             return out
         return impl
+
+
+# Overload the Generator().standard_exponential() method
+@overload_method(types.NumPyRandomGeneratorType, 'standard_exponential')
+def NumPyRandomGeneratorType_standard_exponential(inst, size=None,
+                                                  dtype=np.float64,
+                                                  method=None):
```

Comment:
Can we determine at compile time the value of method ?, Because we need to choose the function accordingly down the line. I put in a very hacky solution by differentiated it on basis of types, None vs string-type.